lib/dashlib.py

```diff
@@ -151,12 +162,18 @@ def create_superblock(proposals, event_block_height, budget_max, sb_epoch_time):
             )
         )
 
-        # else add proposal and keep track of total budget allocation
-        budget_allocated += proposal.payment_amount
-
         payment = {'address': proposal.payment_address,
                    'amount': "{0:.8f}".format(proposal.payment_amount),
                    'proposal': "{}".format(proposal.object_hash)}
+
+        data_length += 2 * (len(payment['address']) + len(payment['amount']) + len(payment['proposal']) + 3)
+
+        if data_length > MAX_GOVERNANCE_OBJECT_DATA_SIZE:
```

Comment:
This is making assumptions about the serialization, right?

It's possible that could change -- why not create a new SB object with each additional proposal, and serialize it to check the length?

Reply:
Right. I serialize only empty SB object and then add lengths of fields for each payment in order to minimize CPU expenses. What about to add additional serialization for each object and check that it's length matches calculated one but only in testnet?